src/plugins/gps_importer/qgsgpsplugin.cpp

```diff
@@ -422,76 +411,77 @@ void QgsGpsPlugin::setupBabel()
+    new QgsBabelSimpleImportFormat( QStringLiteral( "nmea" ), Qgis::BabelFormatCapability::Waypoints | Qgis::BabelFormatCapability::Tracks );
   mImporters[QStringLiteral( "Delorme Street Atlas 2004 Plus" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "saplus" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "saplus" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "Microsoft Streets and Trips" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "s_and_t" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "s_and_t" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "NIMA/GNIS Geographic Names" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "nima" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "nima" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "Maptech" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "mxf" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "mxf" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "Mapopolis.com Mapconverter Application" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "mapconverter" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "mapconverter" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "GPSman" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "gpsman" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "gpsman" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "GPSDrive" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "gpsdrive" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "gpsdrive" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "Fugawi" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "fugawi" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "fugawi" ), Qgis::BabelFormatCapability::Waypoints );
   mImporters[QStringLiteral( "DNA" )] =
-    new QgsSimpleBabelFormat( QStringLiteral( "dna" ), true, false, false );
+    new QgsBabelSimpleImportFormat( QStringLiteral( "dna" ), Qgis::BabelFormatCapability::Waypoints );
 
   // and the GPS devices
   mDevices[QStringLiteral( "Garmin serial" )] =
-    new QgsGpsDevice( QStringLiteral( "%babel -w -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -w -i gpx -o garmin %in %out" ),
-                      QStringLiteral( "%babel -r -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -r -i gpx -o garmin %in %out" ),
-                      QStringLiteral( "%babel -t -i garmin -o gpx %in %out" ),
-                      QStringLiteral( "%babel -t -i gpx -o garmin %in %out" ) );
+    new QgsBabelGpsDeviceFormat( QStringLiteral( "%babel -w -i garmin -o gpx %in %out" ),
```

Comment:
Maybe replace '-w' with Qgis::GpsFeatureType::Waypoint? same for '-t', '-r'

Reply:
This is a string though - i don't understand how it could be used here

Reply:
it's not possible with `QgsAbstractBabelFormat::featureTypeToArgument`?

Reply:
I'll do that in a follow up PR -- this code is being moved shortly

Reply:
Actually I tried this change, but it ended up much more verbose and less readable as a result

Reply:
You're right. It's ok for me.